test/HW04Tests.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE NondecreasingIndentation #-}
+
+module HW04Tests (allTests) where
+
+import Test.Hspec
+
+import Week04.HW04
+
+allTests :: Spec
+allTests = describe "Week04" $ do
+    describe "\nPoly Eq Instance" $ do
+        it "Identical Poly " $ do
+            P ([1, 2] :: [Int]) `shouldBe` P ([1, 2, 0] :: [Int])
```

Comment:
Adding type annotations everywhere seems like an overkill.
You can take advantage of type inference and only add type annotation to the first element of one of the list (and the rest is inferred by the compiler). It's less typing and fewer places to change if you decide to change the representation.
```suggestion
            P [1 :: Int, 2] `shouldBe` P [1, 2, 0]
```